File: 735-asteroid-collision/asteroid-collision.py
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        n = len(asteroids)
        lst = []

        for i in range(0,n):
            
            if asteroids[i]>0:
                lst.append(asteroids[i])
            
            else :
                while (lst and lst[-1]> 0 and lst[-1] < abs(asteroids[i])):
                    lst.pop()

                if (lst and lst[-1]==abs(asteroids[i])):
                    lst.pop()
                
                elif (not lst or lst[-1]<0):
                    lst.append(asteroids[i])
        
        return lst

File: 735-asteroid-collision/asteroid-collision.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        int n = asteroids.length;

        List<Integer> lst = new ArrayList<>();

        for(int i=0; i<n; i++){
            if(asteroids[i]>0) lst.add(asteroids[i]);
            else{
                while(!lst.isEmpty() && lst.get(lst.size()-1)>0 && lst.get(lst.size()-1)<Math.abs(asteroids[i])) {
                    lst.remove(lst.size()-1);
                }

                if(!lst.isEmpty() && lst.get(lst.size()-1)== Math.abs(asteroids[i])) lst.remove(lst.size()-1);

                else if(lst.isEmpty() || lst.get(lst.size()-1)<0) lst.add(asteroids[i]);
            }
        }

        int[] res = new int[lst.size()];

        for(int i=0; i<lst.size(); i++){
            res[i] = lst.get(i);
        }

        return res;
    }
}